sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -864,7 +867,7 @@ def run_GroupByKey(self, transform_node, options):
     windowing = transform_node.transform.get_windowing(transform_node.inputs)
     step.add_property(
         PropertyNames.SERIALIZED_FN,
-        self.serialize_windowing_strategy(windowing))
+        self.serialize_windowing_strategy(windowing, self._default_environment))
```

Comment:
Can we always assume default environment here ? What about x-lang WindowingStrategies ? Probably this can be tested by running a x-lang GBK test with Dataflow with a WindowingStrategy.

Reply:
No, we can't assume this. I don't know where this is being consumed.

Reply:
Heejong clarified that what we pass here is just the default.

Reply:
But the default might not be the right thing here if this is from another language, right? I'd like to better understand where this is being used and what can go wrong if it's missing/incorrect.

Reply:
We simply use the environment passed here as the default environment for PipelineContext that is used when building the windowing proto: https://github.com/apache/beam/blob/master/sdks/python/apache_beam/runners/dataflow/dataflow_runner.py#L1411

I think x-lang custom windowing fn does not work today. Heejong filed https://issues.apache.org/jira/browse/BEAM-10507. I'll look into this but I think this is less severe and does not affect Kafka.

Reply:
@robertwb The only problematic scenario is when `WindowingStrategy` comes from different SDK and the proto doesn't have a value in `environment_id` field. In that case, the host SDK will populate the field with its default environment. But I think it shouldn't happen since it's expansion service's responsibility to pass a semantically correct proto.